ehcache-impl/src/main/java/org/ehcache/impl/persistence/DefaultLocalPersistenceService.java

```diff
@@ -82,6 +85,9 @@ public synchronized void startForMaintenance(ServiceProvider<? super Maintainabl
 
   private void internalStart() {
     if (!started) {
+      if (!rootDirectory.exists() || (!cleanFile.exists() && !lockFile.exists())) {
```

Comment:
I'm think that second condition should really be that the root directory is empty.